executor/aggfuncs/func_group_concat.go

```diff
@@ -491,32 +517,49 @@ func (e *groupConcatDistinctOrder) UpdatePartialResult(sctx sessionctx.Context,
 			continue
 		}
 		p.valSet.Insert(joinedVal)
+		memDelta += int64(len(joinedVal))
 		sortRow := sortRow{
 			buffer:  buffer,
 			byItems: make([]*types.Datum, 0, len(e.byItems)),
 		}
 		for _, byItem := range e.byItems {
 			d, err := byItem.Expr.Eval(row)
 			if err != nil {
-				return 0, err
+				return memDelta, err
 			}
+			memDelta += GetDatumMemSize(&d)
 			sortRow.byItems = append(sortRow.byItems, d.Clone())
 		}
+		oldMem := p.topN.currSize
 		truncated := p.topN.tryToAdd(sortRow)
+		newMem := p.topN.currSize
+		memDelta += int64(newMem - oldMem)
 		if p.topN.err != nil {
-			return 0, p.topN.err
+			return memDelta, p.topN.err
 		}
 		if truncated {
 			if err := e.handleTruncateError(sctx); err != nil {
-				return 0, err
+				return memDelta, err
 			}
 		}
 	}
-	return 0, nil
+	return memDelta, nil
 }
 
 func (e *groupConcatDistinctOrder) MergePartialResult(sctx sessionctx.Context, src, dst PartialResult) (memDelta int64, err error) {
 	// If order by exists, the parallel hash aggregation is forbidden in executorBuilder.buildHashAgg.
 	// So MergePartialResult will not be called.
 	return 0, terror.ClassOptimizer.New(mysql.ErrInternal, mysql.MySQLErrName[mysql.ErrInternal]).GenWithStack("groupConcatDistinctOrder.MergePartialResult should not be called")
 }
+
+func GetDatumMemSize(d *types.Datum) int64 {
+	var datumMemSize int64
+	datumMemSize += int64(unsafe.Sizeof(d.Kind()))
+	datumMemSize += int64(len(d.Collation()))
+	datumValue := d.GetValue()
+	switch datumValue.(type) {
+	case nil, bool, int64, uint64, float64, string, json.BinaryJSON, *types.MyDecimal, []uint8, types.Time, types.Duration:
```

Comment:
IMO, we don't need this switch, because the `getValMemDelta` will do the same thing again.

Reply:
Ok, it is redundant.